core/toolbox/toolbox_item.js

```diff
@@ -0,0 +1,226 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview An item in the toolbox.
+ * @author aschmiedt@google.com (Abby Schmiedt)
+ */
+'use strict';
+
+goog.provide('Blockly.CollapsibleToolboxItem');
+goog.provide('Blockly.SelectableToolboxItem');
+goog.provide('Blockly.ToolboxItem');
+
+
+/**
+ * Class for an item in the toolbox.
+ * @param {!Blockly.utils.toolbox.ToolboxItemDef} toolboxItemDef The JSON defining the
+ *     toolbox item.
+ * @param {!Blockly.IToolbox} toolbox The toolbox that holds the toolbox item.
+ * @constructor
+ * @abstract
+ */
+Blockly.ToolboxItem = function(toolboxItemDef, toolbox) {
+
+  /**
+   * The id for the category.
+   * @type {string}
+   * @protected
+   */
+  this.id_ = toolboxItemDef['id'] || Blockly.utils.genUid();
+
+  /**
+   * The JSON definition of the toolbox item.
+   * @type {!Blockly.utils.toolbox.ToolboxItemDef}
+   * @protected
+   */
+  this.toolboxItemDef_ = toolboxItemDef;
+
+  /**
+   * The toolbox this category belongs to.
+   * @type {!Blockly.IToolbox}
+   * @protected
+   */
+  this.parentToolbox_ = toolbox;
+
+  /**
+   * The workspace of the parent toolbox.
+   * @type {!Blockly.WorkspaceSvg}
+   * @protected
+   */
+  this.workspace_ = this.parentToolbox_.getWorkspace();
+};
+
+/**
+ * Creates the dom for a toolbox item.
+ * @return {?Element} The div for the toolbox item.
```

Comment:
Just a thought, but wouldn't it be best to return a div here, so that we always have createDom returning an `!Element`. If a developer doesn't override this, we end up with a toolbox of empty divs.

Reply:
Yup, I like that better. Thanks!